src/dbms/utils.hpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+#pragma once
+
+#include "dbms/dbms_handler.hpp"
+#include "dbms/replication_handler.hpp"
+#include "replication/include/replication/state.hpp"
+#include "utils/result.hpp"
+
+namespace memgraph::dbms {
+
+inline bool DoReplicaToMainPromotion(dbms::DbmsHandler &dbms_handler) {
+  auto &repl_state = dbms_handler.ReplicationState();
+  // STEP 1) Prepare for new epoch
+  dbms_handler.ForEach([](Database *database) {
+    auto *storage = database->storage();
+    // Remember old epoch + storage timestamp association
+    storage->PrepareForNewEpoch();
+  });
+
+  // STEP 2) Change to MAIN = Kill replication server
+  if (!repl_state.SetReplicationRoleMain()) {
+    return false;
+  }
+
+  // STEP 3) We are now MAIN, update storage local epoch
+  const auto &epoch = std::get<replication::RoleMainData>(std::as_const(repl_state).ReplicationData()).epoch_;
+  dbms_handler.ForEach([&epoch](Database *database) {
+    auto *storage = database->storage();
+    storage->repl_storage_state_.epoch_ = epoch;
+  });
+  return true;
```

Comment:
Still not using the `ReplicationHandler::SetReplicationRoleMain`

Reply:
This is utils method which is used by SetReplicationRoleMain method